### PR TITLE
Forms

### PR DESCRIPTION
**Pull request by TomWoodward**
this copies the the forms system from ancillary service into this lib

the form system in general is pretty rough cut, but i want to start using it in other places. check out the ladle ui for examples

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -19,8 +19,10 @@ module.exports = {
     "import"
   ],
   "rules": {
+    "@typescript-eslint/no-explicit-any": "off",
```
Comment (jivey) [inline .eslintrc.js:22]: I generally like having this as the exception but I see that would add a ton of disable line rules. Wish there was some way to say "in these modules".

Comment (jivey) [reply .eslintrc.js:22]: I forgot about the top of the file one, maybe that would work better if we want to keep things generally strict.

**Comment by jivey** [comment]
@TomWoodward Does the ladle view render for you? I think one of the new package dependencies is crashing the view for me:
```
Uncaught ReferenceError: global is not defined
    at node_modules/buffer/index.js (index.js:43:30)
```

File: .eslintrc.js (hunk 1 above)
Comment (TomWoodward) [reply .eslintrc.js:22]: the no-implicit-any rule makes sense to me, but explicitly specifying `any` is sometimes a necessary escape hatch, and seeing the explicit `any` type is already like a giant neon sign that something weird is going on in the code. i always stop and think twice whenever i see `any` on a PR changes, i just don't think the linter rule is providing any value there

**Comment by TomWoodward** [comment]
@jivey sorry, i guess i hadn't tested ladle since switching out uuid for crypto

**Review by jivey — APPROVED** [review]
This seems like a good basic setup if we only use it internally, but we should make sure to address accessibility before using it anywhere user-facing. I almost want to suggest namespacing it `internal` so we remember to update it. I'm also wondering about visual styles... I see some basic stuff in `uncontrolled`, but it seems likely that we eventually want to make these match the UX library so that consuming apps only have to add custom styles for layout. The unstyled elements might be nice for flexibility, maybe we make the styled ones a seperate export.

Just documenting here while it's fresh in my mind, maybe we use this to make a card for future work:

- make sure aria attributes are all good
- keyboard access and visual state indicators for sortable
- add the remaining types to ladle
- add styles

**Comment by TomWoodward** [comment]
yea thats a good point about the accessibility issues on the sortable. in general i don't think this is ready to use in a public way but i would like for it to get there. adding styles would be a great start in that direction

**Pull request merged**